### PR TITLE
[Docs] Add docs for the task level config

### PR DESCRIPTION
**Pull request by Michaelvll**
Add short docs for #3689

Tested (run the relevant ones):

- [ ] Code formatting: `bash format.sh`
- [ ] Any manual or new tests for this PR (please specify below)
- [ ] All smoke tests: `pytest tests/test_smoke.py` 
- [ ] Relevant individual smoke tests: `pytest tests/test_smoke.py::test_fill_in_the_name` 
- [ ] Backward compatibility tests: `conda deactivate; bash -i tests/backward_compatibility_tests.sh`

### CODE REVIEW

File: docs/source/reference/yaml-spec.rst [hunk 1]
```diff
@@ -331,3 +331,28 @@ Available fields:
 
       # Demoing env var usage.
       echo Env var MODEL_SIZE has value: ${MODEL_SIZE}
+
+Experimental
+------------
+
+In additional to the above fields, SkyPilot also supports the following experimental fields in the task YAML:
```
Comment (romilbhardwaj) [inline docs/source/reference/yaml-spec.rst:338]: Should we add a note box saying something along the lines of: "Experimental features and API may be changed or removed without any notice."

File: docs/source/reference/yaml-spec.rst [hunk 2]
```diff
@@ -331,3 +331,28 @@ Available fields:
 
       # Demoing env var usage.
       echo Env var MODEL_SIZE has value: ${MODEL_SIZE}
+
+Experimental
```
Comment (romilbhardwaj) [inline docs/source/reference/yaml-spec.rst:335]: Should we also add a tip box to config.rst linking to this page? Something like "You can also override some config fields on a per-task basis through the [experimental `config_overrides` field](link)"

**Review by romilbhardwaj — APPROVED** [review]
Thanks @Michaelvll!

File: docs/source/reference/yaml-spec.rst (hunk 1 above)
Comment (Michaelvll) [reply docs/source/reference/yaml-spec.rst:345]: Good idea! Added.

File: docs/source/reference/yaml-spec.rst (hunk 2 above)
Comment (Michaelvll) [reply docs/source/reference/yaml-spec.rst:338]: Good idea. Added.

**Pull request merged**